examples/parse_args.h

```diff
@@ -0,0 +1,211 @@
+    }
+    return NULL;
+}
+
+/**
+ * Check if any options remains in `argv`. Must be called after all expected options are parsed
+ * @param argc
+ * @param argv
+ * @returns NULL if no option was found, else the first option string that was found
+ */
+const char* check_unknown_opts(int argc, char** argv) {
+    for (int i = 1; i < argc; i++) {
+        if (argv[i] && argv[i][0] == '-') {
+            return argv[i];
+        }
+    }
+    return NULL;
+}
+
+/**
+ * Parse positional arguments from `argv`. Must be called after all expected options are parsed, and after checking that
+ * no unknown options remain in `argv`
+ * @param argc
+ * @param argv
+ * @param nb_args: number of expected positional arguments
+ * @returns NULL if found more positional arguments than `nb_args`. Else an array of found arguments in order, followed
+ * by NULL values if found less positional arguments than `nb_args`
+ * @note Returned pointer is dynamically allocated and must be freed
+ */
+char** parse_pos_args(const int argc, char** argv, const size_t nb_args) {
+    char** pos_argv = (char**)calloc(nb_args, sizeof(char*));
+    size_t pos_argc = 0;
+    for (int i = 1; i < argc; i++) {
+        if (argv[i]) {
+            pos_argc++;
+            if (pos_argc > nb_args) {
+                free(pos_argv);
+                return NULL;
+            }
+            pos_argv[pos_argc - 1] = argv[i];
+        }
+    }
+    return pos_argv;
+}
+
+/**
+ * Parse zenoh options that require a JSON-serialized list (-e, -l from common args) and add them to
+ * `config`. Prints error message and exits if fails to insert parsed values
+ * @param argc
+ * @param argv
+ * @param opt: option to parse (without `-` or `--` prefix)
+ * @param config: address of an owned zenoh configuration
+ * @param config_key: zenoh configuration key under which the parsed values will be inserted
+ */
+void parse_zenoh_json_list_config(int argc, char** argv, const char* opt, const char* config_key,
+                                  const z_owned_config_t* config) {
+    char* buf = (char*)calloc(1, sizeof(char));
+    const char* value = parse_opt(argc, argv, opt, true);
+    while (value) {
+        size_t len_newbuf = strlen(buf) + strlen(value) + 4; // value + quotes + comma + nullbyte
+        char* newbuf = (char*)malloc(len_newbuf);
+        snprintf(newbuf, len_newbuf, "%s'%s',", buf, value);
+        free(buf);
+        buf = newbuf;
+        value = parse_opt(argc, argv, opt, true);
+    }
+    size_t buflen = strlen(buf);
+    if (buflen > 0) {
+        // remove trailing comma
+        buf[buflen - 1] = '\0';
+        buflen--;
+        // add list delimiters
+        size_t json_list_len = buflen + 3;  // buf + brackets + nullbyte
+        char* json_list = (char*)malloc(json_list_len);
+        snprintf(json_list, json_list_len, "[%s]", buf);
+        free(buf);
+        // insert in config
+        if (zc_config_insert_json(z_loan(*config), config_key, json_list) < 0) {
```

Comment:
Did you test that it is works? 
Help says it must be string, but error message about JSON, and for me it does not work
```
❯ ../target/release/examples/z_get -m client -l "127.0.0.1:5555"
Couldn't insert value `['127.0.0.1:5555']` in configuration at `listen/endpoints`. This is likely because `listen/endpoints` expects a JSON-serialized list of strings
```

Reply:
It's the same syntax in in Zenoh Rust. You need to remove the quotes and add a protocol to your Locator.
ex: `./target/release/examples/z_get -m client -l tcp/127.0.0.1:5555`